idaes/generic_models/properties/core/eos/ceos.py

```diff
@@ -349,7 +349,92 @@ def dens_mol_phase(b, p):
         else:
             raise PropertyNotSupportedError(_invalid_phase_msg(b.name, p))
 
-    # TODO: Need to add functions to calculate cp and cv
+
+    @staticmethod
+    def cp_mol_phase(blk, p):
+        pobj = blk.params.get_phase(p)
+        if not pobj.is_vapor_phase():
```

Comment:
This error is incorrect - cubic EoS works for both liquid and vapour phase (even though we use the Ideal Gas cp value)